library/Zend/Cache/Backend/Rediscluster.php

```diff
@@ -0,0 +1,536 @@
+	{
+		$namespace = (string) $namespace;
+		if ($namespace === '') {
+			return false;
+		}
+		$keys = $this->_redis->keys($namespace . '*');
+		return $keys;
+	}
+
+	/**
+	 * Get Values of given namespace
+	 *
+	 * @param string $namespace
+	 * @return bool
+	 */
+	public function getValuesByNamespace($namespace)
+	{
+		$namespace = (string) $namespace;
+		if ($namespace === '') {
+			return false;
+		}
+		$keys = $this->_redis->keys($namespace . '*');
+		$values = [];
+		foreach ($keys as $key) {
+			$values[$key] = $this->load($key);
+		}
+		return $values;
+	}
+
+	/**
+	 * Removes TTL from a key
+	 *
+	 * @param string $id
+	 * @return bool
+	 */
+	public function removeTTL($id)
+	{
+		return (boolean) $this->_redis->persist($id);
+	}
+
+	/**
+	 * Return the filling percentage of the backend storage
+	 *
+	 * @throws Zend_Cache_Exception
+	 * @return int integer between 0 and 100
+	 */
+	public function getFillingPercentage()
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return [];
+	}
+
+	/**
+	 * Return an array of stored cache ids
+	 *
+	 * @return array array of stored cache ids (string)
+	 */
+	public function getIds()
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return [];
+	}
+
+	/**
+	 * Return an array of stored tags
+	 *
+	 * @return array array of stored tags (string)
+	 */
+	public function getTags()
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return [];
+	}
+
+	/**
+	 * Return an array of stored cache ids which match given tags
+	 *
+	 * In case of multiple tags, a logical AND is made between tags
+	 *
+	 * @param array $tags array of tags
+	 * @return array array of matching cache ids (string)
+	 */
+	public function getIdsMatchingTags($tags = [])
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return array();
+	}
+
+	/**
+	 * Return an array of stored cache ids which don't match given tags
+	 *
+	 * In case of multiple tags, a logical OR is made between tags
+	 *
+	 * @param array $tags array of tags
+	 * @return array array of not matching cache ids (string)
+	 */
+	public function getIdsNotMatchingTags($tags = [])
+	{
+		$this->_log(self::METHOD_UNSUPPORTED_BY_REDIS_BACKEND);
+		return [];
+	}
+
+	/**
+	 * Return an array of stored cache ids which match any given tags
+	 *
+	 * In case of multiple tags, a logical AND is made between tags
+	 *
+	 * @param array $tags array of tags
+	 * @return array array of any matching cache ids (string)
+	 */
+	public function getIdsMatchingAnyTags($tags = array())
+	{
+		if (!$this->_redis)
+			return [];
+
+		if (!$tags)
+			return [];
+		if ($tags && is_string($tags))
+			$tags = [$tags];
+
+		$return = [];
+		foreach ($tags as $tag) {
+			foreach ($this->_redis->sMembers($this->_keyFromTag($tag)) as $id) {
+				$return[] = $id;
+			}
+		}
+		return $return;
+	}
+
+	/**
+	 * Return the value if is numeric or false if not
+	 *
+	 *
+	 * @param string $id
+	 * @return integer or boolean
```

Comment:
Sorry @ngiallelis my man, I left something back for you, just to remind you the goodDays